### ISSUE TITLE
Knowledge - Ingestion fails with "timeout waiting for RunState <runid> to meeting condition"

### ISSUE THREAD

**Issue by sangee2004** [obot-platform/obot#461]
**Steps to reproduce the problem:**

1. Create a knowledge file from notion - "My new test page"
2. Once file syncing is done, select all files for ingesting.

Ingesting 3 of files fail with following errors:

<img width="1081" alt="Screenshot 2024-11-06 at 10 56 28 AM" src="https://github.com/user-attachments/assets/2308950b-ba38-4486-986d-8def8ced552f">

When trying to ingest the same set of files again, 1 of the files failed with same error
<img width="1190" alt="Screenshot 2024-11-06 at 11 34 51 AM" src="https://github.com/user-attachments/assets/46165cd4-9342-4ac5-97ff-a9269cbbe6fd">

Ingestion error details:

```

  {
      "id": "33592a9a05149cedeaeb93a71836ec250af606c0a317a00963530def6b51c8a6",
      "created": "2024-11-06T17:37:50Z",
      "fileName": "126b081d-a8af-81a7-9c52-fa34d4da5b07/My check1.md",
      "state": "error",
      "error": "failed to ingest file: timeout waiting for RunState r1fl895 to meet condition",
      "agentID": "a17zt6p",
      "knowledgeSetID": "kst1-a17zt6p",
      "knowledgeSourceID": "ks14b5dj",
      "approved": true,
      "url": "https://www.notion.so/My-check1-126b081da8af81a79c52fa34d4da5b07",
      "updatedAt": "2024-10-21T19:21:00.000Z",
      "lastIngestionStartTime": "2024-11-06T19:27:05Z",
      "lastIngestionEndTime": "2024-11-06T19:29:43Z",
      "lastRunIDs": [
        "r1fdsrk",
        "r1fl895"
      ]
    }
```
**Note** - This issue may not be reproducible all the time.
Able to ingest the file successfully on re ingestion.

**Comment by iwilltry42** [comment]
@sangee2004 did you see a similar issue recently?

**Comment by sangee2004** [comment]
Have not encountered this ingestion error for sometime now.

Closing this issue. If i see a similar issue happen again, will log a new issue to track it.

**Comment by sangee2004** [comment]
I am seeing this issue in test env when trying to ingest all files from intel.com

Agent - https://test.acornlabs.com/admin/agents/a1jnvkh

<img width="1056" alt="Screenshot 2025-01-08 at 2 29 41 PM" src="https://github.com/user-attachments/assets/f12c17af-4ff4-40c0-a3b9-732233255f33" />
